hack/assets/assets.go

```diff
@@ -0,0 +1,214 @@
+		fmt.Printf("Invalid value for --kubernetes-distro. Accepted values are: %s", strings.Join(GetSupportedDistros(), ", "))
+		os.Exit(1)
+	}
+
+	if *kubernetesVersion != "" && !slices.Contains(k8sSupportedVersions, *kubernetesVersion) {
+		fmt.Printf("Invalid value for --kubernetes-version. Accepted values are: %s", strings.Join(k8sSupportedVersions, ", "))
+		os.Exit(1)
+	}
+
+	if *latest && *kubernetesVersion != "" {
+		fmt.Println("Flags --latest and --kubernetes-version are not compatible")
+		os.Exit(1)
+	}
+
+	if *latest && *optional {
+		fmt.Println("Flags --latest and --optional are not compatible")
+		os.Exit(1)
+	}
+
+	cleanTag := strings.TrimLeft(pflag.Arg(0), "v")
+
+	images := GetImages(cleanTag, *latest, *proOnly, *optional, *kubernetesVersion, *kubernetesDistro)
+	for _, img := range images {
+		fmt.Println(img)
+	}
+}
+
+// GetSupportedDistros returns a list of supported Kubernetes distros
+func GetSupportedDistros() []string {
+	return []string{k8s, k3s, k0s}
+}
+
+// GetImages returns a list of images based on the given parameters
+func GetImages(cleanTag string, latest bool, proOnly bool, optional bool, kubernetesVersion string, kubernetesDistro string) []string {
+	var images []string
+	if !optional {
+		images = GetVclusterProImages(cleanTag)
+		if !proOnly {
+			images = append([]string{"ghcr.io/loft-sh/vcluster:" + cleanTag}, images...) // Add oss image
+		}
+	}
+	images = UniqueAppend(images,
+		GetImageList(latest, optional, kubernetesVersion, GetVclusterDependencyImageMaps(kubernetesDistro))...,
+	)
+	return images
+}
+
+// GetSupportedKubernetesVersions returns a list of supported Kubernetes versions
+func GetSupportedKubernetesVersions() []string {
+	k8sSupportedVersions := maps.Keys(vclusterconfig.K8SAPIVersionMap)
+	slices.SortFunc(k8sSupportedVersions, versionsDescCmp)
+	return k8sSupportedVersions
+}
+
+// GetImageList returns a list of images based on the given groups
+// If latest is true, only the latest image of each group is returned
+// If kubernetesVersion is specified, only the images matching the version are returned
+func GetImageList(latest bool, optional bool, kubernetesVersion string, groups []map[string]string) []string {
+	selectedImages := make([]string, 0, len(groups))
+	for _, g := range groups {
+		if len(g) == 0 {
+			continue
+		}
+		if kubernetesVersion != "" {
+			if img, ok := g[kubernetesVersion]; ok {
+				selectedImages = append(selectedImages, img)
+			}
+			continue
+		}
+		sortedImages := slices.Compact(getSortedDescValues(g))
+		if latest {
+			selectedImages = append(selectedImages, sortedImages[0])
+			continue
+		}
+		if optional {
+			selectedImages = append(selectedImages, sortedImages[1:]...)
+			continue
+		}
+		selectedImages = append(selectedImages, sortedImages[:]...)
+	}
+	return selectedImages
+}
+
+// GetVclusterProImages returns a list of vcluster images
+func GetVclusterProImages(cleanTag string) []string {
```

Comment:
> The -pro image should stay for vcluster, while the vcluster one should be removed and the vcluster-oss one should be included on optional one

I oddly don't see the normal one here ? But it's getting generated somewhere. We also need the `-oss` one.

Reply:
Done